### ISSUE TITLE
Streaming Helper `writeSSE()` stuck when connection closed

### ISSUE THREAD

**Issue by Tsuk1ko** [honojs/hono#1902]
### What version of Hono are you using?

3.12.0

### What runtime/platform is your app running on?

Node v18.19.0

### What steps can reproduce the bug?

Here is the minimum reproduction code:

```ts
import { serve } from '@hono/node-server';
import { Hono } from 'hono';
import { streamSSE } from 'hono/streaming';

const app = new Hono();
app.get('/', c =>
  streamSSE(c, async stream => {
    while (true) {
      const now = Date.now();
      console.log('write start', now);
      await stream.writeSSE({ data: String(now) }).finally(() => {
        console.log('write end', now);
      });
      await stream.sleep(1000);
    }
  }),
);

serve(app);
```

Access http://localhost:3000/ then stop page loading (or just close it).

### What is the expected behavior?

If connection closed, it should throw an error or provide a handler to do sth.

### What do you see instead?

The loop will continue to run for a while and then get stuck at `streamSSE()`, no error was thrown.

Page stop at here:

![image](https://github.com/honojs/hono/assets/24877906/3ff8664d-f724-449f-a361-fa37ab85e321)

Log stop at here:

![image](https://github.com/honojs/hono/assets/24877906/502f6a0c-5fb9-42c4-97d1-ddcdd608e4ab)

### Additional information

When using hono@3.11.12 it will close the stream but no error thrown out (will be throw internally and log a message), so I handled it by rewriting the close function:

```ts
const handleStreamClose = (stream: StreamingApi, onClose: () => any) => {
  const close = stream.close.bind(stream);
  stream.close = async () => {
    try {
      await onClose();
    } finally {
      await close();
    }
  };
};
```

But on 3.12.0 the stream won't be close so this don't work anymore, `stream.onAbort()` isn't triggered either.

Is there a better solution to handle this situation?

**Comment by watany-dev** [comment]
I have confirmed that this bug is reproduced, I have made some light changes on the helper side but it may not be detecting the disconnection from the client.
I would also like to have the version information of the nodejs adaptor just in case. I'm considering the possibility that we may need to make some changes here.

**Comment by yusukebe** [comment]
@Tsuk1ko Thank you for the issue!

@sor4chi Could you take a look this?

**Comment by watany-dev** [comment]
Thank you @sor4chi !

@Tsuk1ko 
Once this patch is in, you will be able to stop it with onAbort

https://github.com/honojs/hono/pull/1920

```ts
import { serve } from '@hono/node-server';
import { Hono } from 'hono';
import { streamSSE } from 'hono/streaming';

const app = new Hono();
app.get('/', c =>
  streamSSE(c, async stream => {
    let isAborted: boolean = false
    stream.onAbort(() => {
      console.log("abort!!!!")
      isAborted = true
    })
    while (!isAborted) {
      const now = Date.now();
      console.log('write start', now);
      await stream.writeSSE({ data: String(now) }).finally(() => {
        console.log('write end', now);
      });
      await stream.sleep(1000);
    }
  }),
);

serve(app);
```

**Comment by Tsuk1ko** [comment]
Thank you!

**Comment by Tsuk1ko** [comment]
Confirmed to be fixed by v3.12.1 👍
